--- a/target/linux/mediatek/files-6.12/drivers/net/wireless/wifi_utility/pci_mediatek_rbus.c
+++ b/target/linux/mediatek/files-6.12/drivers/net/wireless/wifi_utility/pci_mediatek_rbus.c
@@ -136,7 +136,7 @@ rbus_pa_lan_config(struct platform_device *pdev, unsigned int devfn, unsigned ch
 {
 	struct pinctrl *p;
 	struct pinctrl_state *s;
-	unsigned char state[32] = "";
+	const char *state = "state_epa";
 	int ret = 0;
 
 	if (mode != IPA_ELNA_MODE && mode != EPA_ELNA_MODE)
@@ -150,8 +150,6 @@ rbus_pa_lan_config(struct platform_device *pdev, unsigned int devfn, unsigned ch
 		return ret;
 	}
 
-	strncpy(state, "state_epa", sizeof("state_epa"));
-
 	s = pinctrl_lookup_state(p, state);
 
 	if (!s) {
